python/F3DPythonBindings.cxx

```diff
@@ -156,39 +156,33 @@ PYBIND11_MODULE(pyf3d, module)
 
   options //
     .def(py::init<>())
-    .def("__setitem__", py::overload_cast<const std::string&, bool>(&f3d::options::set))
-    .def("__setitem__", py::overload_cast<const std::string&, int>(&f3d::options::set))
-    .def("__setitem__", py::overload_cast<const std::string&, double>(&f3d::options::set))
-    .def(
-      "__setitem__", py::overload_cast<const std::string&, const std::string&>(&f3d::options::set))
-    .def("__setitem__",
-      py::overload_cast<const std::string&, const std::vector<int>&>(&f3d::options::set))
     .def("__setitem__",
-      py::overload_cast<const std::string&, const std::vector<double>&>(&f3d::options::set))
+      [](f3d::options& opts, const std::string& key, const f3d::option_variant_t& value)
+      {
+        try
+        {
+          opts.set(key, value);
+        }
+        catch (const f3d::options::inexistent_exception&)
```

Comment:
Might be better to use exception translators instead:
```cpp
py::register_local_exception<f3d::options::inexistent_exception>(module, "F3DInexistentOption", PyExc_KeyError);
```

See https://pybind11.readthedocs.io/en/stable/advanced/exceptions.html#registering-custom-translators

Reply:
Ill create an issue for this one, I'd rather not touch the python bindings logic too much if possible

Reply:
ok, then you can resolve the discussion when the issue is opened